firefox-ios/Storage/Rust/RustFirefoxSuggest.swift

```diff
@@ -42,22 +50,44 @@ public actor RustFirefoxSuggest: RustFirefoxSuggestActor {
         // downloading new suggestions. This is safe to call multiple times.
         Viaduct.shared.useReqwestBackend()
 
-        try store.ingest(constraints: SuggestIngestionConstraints())
+        try await withCheckedThrowingContinuation { continuation in
+            writerQueue.async(qos: .utility) { [weak self] in
+                do {
+                    try self?.store.ingest(constraints: SuggestIngestionConstraints())
+                    continuation.resume()
+                } catch {
+                    continuation.resume(throwing: error)
+                }
```

Comment:
Continuations can be a little sensitive so just want to confirm, are you very confident store.ingest will return exactly once? Same for the other continuation below.
For context if a continuation is resumed more than once or if it's never resumed the app crashes.

Reply:
> confirm, are you very confident store.ingest will return exactly once?

Yes.

`store.ingest()` is a regular blocking function that can only return once.

`DispatchQueue.async` is also guaranteed to only run the block once, unless cancelled. The block captures a strong reference to `self`, and we never explicitly cancel any work that we dispatch to the queue; IIUC, this guarantees the block will be called.